src/parser/scalar.js

```diff
@@ -335,12 +339,14 @@ module.exports = {
       result = result(
         "string",
         false,
-        this.remove_heredoc_leading_whitespace_chars(
-          this.resolve_special_chars(text, isDoubleQuote),
-          this.lexer.heredoc_label.indentation,
-          this.lexer.heredoc_label.indentation_uses_spaces,
-          this.lexer.heredoc_label.first_encaps_node
-        ),
+        this.version >= 703 && !this.lexer.heredoc_label.finished
```

Comment:
There is a bug where if you have a heredoc with an indented ending followed by a string with newlines, it will trigger an error if the string's newline isn't indented as much as the heredoc's ending.  I use the `heredoc_label.finished` variable to mark when the parser has finished parsing the heredoc block and does not have to follow the indentation rules any longer.